src/Amqp/JobQueue/DeclarationManager.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Icecave\Overpass\Amqp\JobQueue;
+
+use PhpAmqpLib\Channel\AMQPChannel;
+
+/**
+ * @access private
+ */
+class DeclarationManager
+{
+    public function __construct(AMQPChannel $channel)
+    {
+        $this->channel = $channel;
+        $this->jobQueues = [];
+    }
+
+    /**
+     * @return string
+     */
+    public function exchange()
+    {
+        if ($this->exchange) {
+            return $this->exchange;
+        }
+
+        $name = 'overpass/job';
+
+        $this
+            ->channel
+            ->exchange_declare(
+                $name,
+                'direct',
+                false, // passive,
+                false, // durable,
+                false  // auto delete
+            );
+
+        $this->exchange = $name;
+
+        return $this->exchange;
+    }
+
+    /**
+     * @param string $task
+     *
+     * @return string
+     */
+    public function jobQueue($task)
+    {
+        if (isset($this->jobQueues[$task])) {
+            return $this->jobQueues[$task];
+        }
+
+        list($queueName) = $this
+            ->channel
+            ->queue_declare(
+                'overpass/job/' . $task,
+                false, // passive
+                true, // durable
```

Comment:
Changed my mind about this for now - let's leave `durable` as false until we have a use case for persisted messages.